actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -82,11 +81,19 @@ def format(formatter, filter = nil)
 
       private
 
-      def filter_routes(filter)
-        if filter
+      def filter_routes(options)
+        filter = filter(options)
+        filtered_routes = if filter
           @routes.select { |route| route.defaults[:controller] == filter }
-        else
-          @routes
+        end
+
+        (filtered_routes || @routes).select { |r| /#{options[:pattern]}/ =~ r.defaults.values.join(' ') }
```

Comment:
`/#{options[:pattern]}/` could be expensive for many routes, can we extract it to a local variable?